src/engine/effects/engineeffectsdelay.h

```diff
@@ -76,6 +78,6 @@ class EngineEffectsDelay final : public EngineObject {
   private:
     SINT m_currentDelaySamples;
     SINT m_prevDelaySamples;
-    SINT m_delayBufferWritePos;
-    CSAMPLE* m_pDelayBuffer;
+    mixxx::SampleBuffer m_currentDelayBuffer;
+    std::unique_ptr<RingDelayBuffer> m_pDelayBuffer;
```

Comment:
I don't see any reason for allocating here. RingDelayBuffer already does internally. 
```suggestion
    RingDelayBuffer m_pDelayBuffer;
```